src/Universe/EntityRepository.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EtoA\Universe;
+
+use EtoA\Core\AbstractRepository;
+
+class EntityRepository extends AbstractRepository
+{
+    public function findRandomId(string $code): ?int
+    {
+        $id = $this->createQueryBuilder()
+            ->select('id')
+            ->from('entities')
+            ->where('code = :code')
+            ->orderBy('RAND()')
+            ->setParameters([
+                'code' => $code,
+            ])
+            ->execute()
+            ->fetchOne();
+
+        return $id !== false ? (int) $id : null;
+    }
+
+    public function findRandomByCodes(array $codes, int $limit): array
+    {
+        if (count($codes) == 0) {
+            return [];
+        }
+
+        return $this->createQueryBuilder()
+            ->select('*')
+            ->from('entities')
+            ->where('code IN (' . implode(',', array_fill(0, count($codes), '?')) . ')')
+            ->andWhere('pos = 0')
+            ->orderBy('RAND()')
+            ->setParameters(array_values($codes))
+            ->setMaxResults($limit)
+            ->execute()
+            ->fetchAllAssociative();
+    }
+
+    public function findIncludeCell(int $id): ?array
+    {
+        $data = $this->createQueryBuilder()
+            ->select(
+                'e.id',
+                'c.id as cid',
+                'code',
+                'pos',
+                'sx',
+                'sy',
+                'cx',
+                'cy'
+            )
+            ->from('entities', 'e')
+            ->innerJoin('e', 'cells', 'c', 'e.cell_id = c.id')
+            ->where('e.id = :id')
+            ->setParameters([
+                'id' => $id,
+            ])
+            ->execute()
+            ->fetchAssociative();
+
+        return $data !== false ? $data : null;
```

Comment:
This method and the method below could use the same object definition

Reply:
Done 
4742666